test/libsolidity/syntaxTests/inlineAssembly/evm_constantinople_on_byzantium.sol

```diff
@@ -0,0 +1,32 @@
+contract C {
+    function f() view external returns (uint ret) {
+        assembly {
+            ret := shl(gas(), 5)
+            ret := shr(ret, 2)
+            ret := sar(ret, 2)
+        }
+    }
+    function g() external returns (address ret) {
+        assembly {
+            ret := create2(0, 0, 0, 0)
+        }
+    }
+    function h() pure external returns (uint ret) {
+        assembly {
+            ret := chainid()
+        }
+    }
+}
+// ====
+// EVMVersion: =byzantium
+// ----
+// TypeError 6612: (103-106): The "shl" instruction is only available for Constantinople-compatible VMs (you are currently compiling for "byzantium").
+// DeclarationError 8678: (96-116): Variable count does not match number of values (1 vs. 0)
+// TypeError 6612: (136-139): The "shr" instruction is only available for Constantinople-compatible VMs (you are currently compiling for "byzantium").
+// DeclarationError 8678: (129-147): Variable count does not match number of values (1 vs. 0)
+// TypeError 6612: (167-170): The "sar" instruction is only available for Constantinople-compatible VMs (you are currently compiling for "byzantium").
```

Comment:
This raises the question: should we split these cases in `validateInstruction`? :)